src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs

```diff
@@ -0,0 +1,349 @@
+
+        /// <summary>
+        /// InputStream parameter
+        /// The stream of the file to calculate a hash
+        /// </summary>
+        /// <value></value>
+        [Parameter(Mandatory = true, ParameterSetName = StreamParameterSet, Position = 0)]
+        public Stream InputStream { get; set; }
+
+        /// <summary>
+        /// BeginProcessing() override
+        /// This is for hash function init
+        /// </summary>
+        protected override void BeginProcessing()
+        {
+            InitHasher(Algorithm);
+
+            if (ParameterSetName == StreamParameterSet)
+            {
+                byte[] bytehash = null;
+                String hash = null;
+
+                bytehash = hasher.ComputeHash(InputStream);
+
+                hash = BitConverter.ToString(bytehash).Replace("-","");
+                WriteHashResult(Algorithm, hash, "");
+
+                return;
+            }
+
+        }
+
+        /// <summary>
+        /// ProcessRecord() override
+        /// This is for paths collecting from pipe
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            List<string> pathsToProcess = new List<string>();
+            ProviderInfo provider = null;
+
+            switch (ParameterSetName)
+            {
+                case PathParameterSet:
+                    // Resolve paths and check existence
+                    foreach (string path in _paths)
+                    {
+                        try
+                        {
+                            Collection<string> newPaths = Context.SessionState.Path.GetResolvedProviderPathFromPSPath(path, out provider);
+                            if (newPaths != null)
+                            {
+                                pathsToProcess.AddRange(newPaths);
+                            }
+                        }
+                        catch (ItemNotFoundException e)
+                        {
+                            if (!WildcardPattern.ContainsWildcardCharacters(path))
+                            {
+                                ErrorRecord errorRecord = new ErrorRecord(e,
+                                    "FileNotFound",
+                                    ErrorCategory.ObjectNotFound,
+                                    path);
+                                WriteError(errorRecord);
+                            }
+                        }
+                    }
+                    break;
+                case LiteralPathParameterSet:
+                    foreach (string path in _paths)
+                    {
+                        string newPath = Context.SessionState.Path.GetUnresolvedProviderPathFromPSPath(path);
+
+                        // Unlike 'GetResolvedProviderPathFromPSPath'
+                        // 'GetUnresolvedProviderPathFromPSPath' does not check a file existence
+                        // so do that explicity
+                        if (File.Exists(newPath))
```

Comment:
What I found:
1. `Unblock-File` use the same pattern.
2. For `Path` parameter `GetResolvedProviderPathFromPSPath` has the same problem - "the file might disappear between this check and when we try to read it."

So we could keep consistency here.

Later we try read the file and _ignore_ any exceptions.
The original `Get-FileHash` function call [Write-Error and return](https://github.com/PowerShell/PowerShell/blob/master/src/Modules/Shared/Microsoft.PowerShell.Utility/Microsoft.PowerShell.Utility.psm1#L128)
Should we reproduce the same behavior or  _ignore_ any exceptions?